### PR TITLE
power pc and arm64 fixes

### PR DESCRIPTION
**Pull request by houlz0507**

### CODE REVIEW

File: src/runtime_src/driver/xclng/drm/xocl/mgmtpf/mgmt-core.c [hunk 1]
```diff
@@ -362,7 +362,7 @@ static struct xclmgmt_char *create_char(struct xclmgmt_dev *lro)
 	/* couple the control device file operations to the character device */
 	cdev_init(&lro_char->cdev, &ctrl_fops);
 	lro_char->cdev.owner = THIS_MODULE;
-	lro_char->cdev.dev = MKDEV(MAJOR(xclmgmt_devnode), lro->instance);
+	lro_char->cdev.dev = MKDEV(MAJOR(xclmgmt_devnode), lro->core.dev_minor);
```
Comment (maxzhen) [inline src/runtime_src/driver/xclng/drm/xocl/mgmtpf/mgmt-core.c]: Can we just keep one "id", either "instance" or "core.dev_minor"? They seem to serve the same purpose.

File: src/runtime_src/driver/xclng/drm/xocl/userpf/qdma.c [hunk 2]
```diff
@@ -117,6 +117,10 @@ static int xocl_user_qdma_probe(struct pci_dev *pdev,
 	ocl_dev->core.pdev = pdev;
 	xocl_fill_dsa_priv(ocl_dev, dev_info);
 
+	ret = xocl_alloc_dev_minor(ocl_dev);
```
[the remark this answers is not in the document]
Comment (houlz0507) [reply src/runtime_src/driver/xclng/drm/xocl/userpf/qdma.c]: fixed

File: src/runtime_src/driver/xclng/drm/xocl/mgmtpf/mgmt-core.c [hunk 3]
```diff
@@ -798,6 +802,8 @@ static int xclmgmt_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	unmap_bars(lro);
 err_map:
 	pci_release_regions(pdev);
+err_alloc_minor:
```
[the remark this answers is not in the document]
Comment (houlz0507) [reply src/runtime_src/driver/xclng/drm/xocl/mgmtpf/mgmt-core.c]: fixed

File: src/runtime_src/driver/xclng/drm/xocl/xocl_subdev.c [hunk 4]
```diff
@@ -516,3 +518,26 @@ int xocl_xrt_version_check(xdev_handle_t xdev_hdl,
...
+	}
+
+	return 0;
+}
+
+void xocl_free_dev_minor(xdev_handle_t xdev_hdl)
+{
+	struct xocl_dev_core *core = (struct xocl_dev_core *)xdev_hdl;
+
+	if (core->dev_minor >= 0)
```
[the remark this answers is not in the document]
Comment (houlz0507) [reply src/runtime_src/driver/xclng/drm/xocl/xocl_subdev.c]: uses -1 as invalid minor. alloc_minor happens just after allocate driver data. And init dev_minor to invalid allocation failed in alloc_minor().

File: src/runtime_src/driver/xclng/drm/xocl/mgmtpf/mgmt-core.c (hunk 1 above)
Comment (houlz0507) [reply src/runtime_src/driver/xclng/drm/xocl/mgmtpf/mgmt-core.c]: The maximum minor is 20bits. However, pci id is 32bits which we used as device instance name. e.g. /dev/xclmgmt<instance>. So I would still use pci id as char dev name/sysfs name, but another minor id with is invisible for end user.

**Pull request merged**